src/polymer/polymer2-element-scanner.ts

```diff
@@ -380,7 +380,7 @@ class ClassFinder implements Visitor {
       return;
     }
 
-    const name = node.id && astValue.getIdentifierName(node.id);
+    const name = (node.id && astValue.getIdentifierName(node.id)) || undefined;
```

Comment:
I think I'd prefer `const name = node.id ? astValue.getIdentifierName(node.id) : undefined;` as more explicit, and shorter than the bool operator version.